Kudu.Core/Infrastructure/FileSystemHelpers.cs

```diff
@@ -130,6 +130,15 @@ public static DateTime GetLastWriteTimeUtc(string path)
             return Instance.File.GetLastWriteTimeUtc(path);
         }
 
+        // Get the latest LastWriteTime of all the files in a folder
+        public static DateTime GetFolderLatestWriteTimeUtc(string path)
+        {
+            DirectoryInfoBase jobBinariesDirectory = DirectoryInfoFromDirectoryName(path);
+            FileInfoBase[] files = jobBinariesDirectory.GetFiles("*.*", SearchOption.AllDirectories);
+
+            return files.Select(f => f.LastWriteTimeUtc).OrderByDescending(t => t).FirstOrDefault();
```

Comment:
@amitapl you are correct, and this leaves some scenarios where we'll fail to update. @mathewc is looking at a somewhat more elaborate fix that you handle those cases. Thanks!